phys2bids/physio_obj.py

```diff
@@ -0,0 +1,400 @@
+    timeseries : (ch x tps) :obj:`numpy.ndarray`
+        Numpy 2d array of timeseries
+        Contains all the timeseries recorded.
+        Impose same frequency!
+    freq : float
+        Shared frequency of the object.
+            Properties
+    ch_name : (ch) list of strings
+        List of names of the channels - can be the header of the columns
+        in the output files.
+    units : (ch) list of strings
+        List of the units of the channels.
+    start_time : float
+        Starting time of acquisition (equivalent to first TR,
+        or to the opposite sign of the time offset).
+
+    Methods
+    -------
+    return_index:
+        Returns the proper list entry of all the
+        properties of the object, given an index.
+    delete_at_index:
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+    init_from_blueprint:
+        method to populate from input blueprint instead of init
+    """
+    def __init__(self, timeseries, freq, ch_name, units, start_time):
+        self.timeseries = is_valid(timeseries, np.ndarray)
+        self.ch_amount = self.timeseries.shape[0]
+        self.freq = is_valid(freq, (int, float))
+        self.ch_name = has_size(ch_name, self.ch_amount, 'unknown')
+        self.units = has_size(units, self.ch_amount, '[]')
+        self.start_time = start_time
+
+    def return_index(self, idx):
+        """
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+
+        Input
+        -----
+        self: :obj: `BlueprintOutput`
+            The object on which to operate
+        idx: int
+            Index of elements to return
+
+        Output
+        ------
+        out: tuple
+            Tuple containing the proper list entry of all the
+            properties of the object with index `idx`
+        """
+        return (self.timeseries[idx], self.ch_amount, self.freq,
+                self.ch_name[idx], self.units[idx], self.start_time)
+
+    def delete_at_index(self, idx):
+        """
+        Returns all the proper list entry of the
+        properties of the object, given an index.
+
+        Input
+        -----
+        self: :obj: `BlueprintOutput`
+            The object on which to operate
+        idx: int or range
+            Index of elements to delete from all lists
+
+        Outcome
+        -------
+        self:
+            In all the property that are lists, the element correspondent to
+            `idx` gets deleted
+        """
+        del(self.timeseries[idx])
+        del(self.ch_name[idx])
+        del(self.units[idx])
+
+    @classmethod
+    def init_from_blueprint(cls, blueprint):
+        """
+        Method to populate the output blueprint using BlueprintInput.
+
+        Input
+        -----
+        cls: :obj: `BlueprintOutput`
+            The object on which to operate
+        blueprint: :obj: `BlueprintInput`
+            The input blueprint object
```

Comment:
This is actually not true, the input here is ``BlueprintInput[selected_frequency]`` no?

Reply:
I don't think so. In `phys2bids.py`, `phys_out` is a dictionary that contains a bunch of `BlueprintOutput` objects. Here, it's the class itself.